### ISSUE TITLE
Test-Json returns Error before result

### ISSUE THREAD

**Issue by alexandru-mocanu** [PowerShell/PowerShell#11999]
If validation is unsuccessful, the cmdlet returns the error before the result. If you want to handle the error returned by the cmdlet you are unable to get to the result. It should return the result first and then the error.

**Comment by mklement0** [comment]
You can handle the non-terminating errors that `Test-Json` returns without aborting execution, so you'll still get the result.

Also, if you choose to abort execution, you don't need the result, because `$false` is implied.

```powershell
# Ignore errors and capture only the result.
$result = Test-Json '][' -ErrorAction Ignore

# Silence the error messages, but capture them in variable $err
$result = Test-Json '][' -ErrorAction SilentlyContinue -ErrorVariable err
```

Generally, in a cmdlet that outputs a single result after overall processing has completed, it's appropriate to emit non-terminating errors that occur in the course of processing  _before_ the result .

**Comment by vexx32** [comment]
The general usage of `Test-*` cmdlets is that they _do not error_. The only other exception currently is `Test-Connection`, primarily because that behaviour exists in utilities like `ping` as well as the original `Test-Connection` and `Test-NetConnection` cmdlets behaved that way.

`Test-Path` for example, pretty much will not error unless you pass it completely invalid input (iirc currently it can only error if you pass it `$null` which fails parameter validation).

If there's an error... it should just return False and nothing else, really.

**Comment by mklement0** [comment]
Good point, @vexx32 - the real problem here is that _errors_ are emitted at all, _by default_, no less.

@iSazonov, is changing that on the radar for your PR #11397?

**Comment by iSazonov** [comment]
There is #11934 where we could discuss and address the issue.

**Comment by gregsdennis** [comment]
The JSON Schema implementation has been recently updated to JsonSchema.Net and is available in the latest Powershell snapshot. Please retest this to see if it is still a problem.

**Comment by ft3411** [comment]
The issue still exists in PowerShell 7.6.3
Could someone have a look at it?
